pkg/services/ngalert/store/alert_rule.go

```diff
@@ -298,8 +300,19 @@ func (st DBstore) GetNamespaceAlertRules(query *ngmodels.ListNamespaceAlertRules
 func (st DBstore) GetRuleGroupAlertRules(query *ngmodels.ListRuleGroupAlertRulesQuery) error {
 	return st.SQLStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
 		alertRules := make([]*ngmodels.AlertRule, 0)
-		q := "SELECT * FROM alert_rule WHERE org_id = ? and namespace_uid = ? and rule_group = ?"
-		if err := sess.SQL(q, query.OrgID, query.NamespaceUID, query.RuleGroup).Find(&alertRules); err != nil {
+
+		s := strings.Builder{}
+		params := make([]interface{}, 0)
+
+		addToQuery := func(stmt string, p ...interface{}) {
+			s.WriteString(stmt)
+			params = append(params, p...)
+		}
+		q := "SELECT * FROM alert_rule WHERE org_id = ? and rule_group = ?"
+		if query.NamespaceUID != "" {
+			addToQuery(` AND namespace_ui = ?`, query.NamespaceUID)
```

Comment:
There is an [index](https://github.com/grafana/grafana/blob/e288cd0836d83b34da37cba14691feac9859b969/pkg/services/ngalert/store/database_mig.go#L137) on org_id, namespace_uid, group_uid columns but I think that if the namespace is missing then it won't be used (in some database at least).
so I think that we have to reverse the column order to  org_id, group_uid, namespace_uid.

But do we really need this? Personally I would prefer to always pass the namespace.
Another reason for that is that there can be groups with the same name under different namespaces.

Reply:
SGTM, thank you!

Reply:
Reverted this change. Thanks for looking at it closely!